xlators/features/marker/src/marker-quota.c

```diff
@@ -1685,9 +1685,7 @@ mq_initiate_quota_task(void *opaque)
             goto out;
         }
 
-        if (prev_dirty == 0) {
-            ret = mq_mark_dirty(this, &parent_loc, 0);
```

Comment:
We still need to cover the case when prev_dirty is zero. What will happen if this condition is true.

Reply:
```
if (prev_dirty == 0) {
            ret = mq_mark_dirty(this, &parent_loc, 0);
        } else {
            ret = mq_inode_ctx_get(parent_loc.inode, this, &parent_ctx);
            if (ret == 0)
                mq_set_ctx_dirty_status(parent_ctx, _gf_false);
        }

        dirty = _gf_false;
        prev_dirty = 0;
```
prev_dirty is made 0 explicitly in the following statements. I guess that handles the scene itself.